common/spec/dependabot/sem_version_spec.rb

```diff
@@ -0,0 +1,215 @@
+
+      it { is_expected.to be(true) }
+    end
+
+    context "with a malformed version" do
+      let(:version_string) { "1.2" }
+
+      it { is_expected.to be(false) }
+    end
+
+    context "with a malformed version" do
+      let(:version_string) { "1.0.0.pre1" }
+
+      it { is_expected.to be(false) }
+    end
+
+    context "with a malformed version" do
+      let(:version_string) { "Finchley" }
+
+      it { is_expected.to be(false) }
+    end
+
+    context "with a malfomed version" do
+      let(:version_string) { "1.+" }
+
+      it { is_expected.to be(false) }
+    end
+
+    context "with an underscore-specified prerelease" do
+      let(:version_string) { "1.0.0_pre1" }
+
+      it { is_expected.to be(false) }
+    end
+
+    context "with an alpha separated with a ." do
+      let(:version_string) { "1.0.0.alpha" }
+
+      it { is_expected.to be(false) }
+    end
+
+    context "with an alpha with no separator" do
+      let(:version_string) { "1.0.0alpha" }
+
+      it { is_expected.to be(false) }
+    end
+
+    context "with a 'pr' pre-release separated with a ." do
+      let(:version_string) { "2.10.0.pr3" }
+
+      it { is_expected.to be(false) }
+    end
+  end
+
+  describe "#prerelease?" do
+    subject { version.prerelease? }
+
+    context "with an alpha" do
+      let(:version_string) { "1.0.0-alpha" }
+
+      it { is_expected.to be(true) }
+    end
+
+    context "with a capitalised alpha" do
+      let(:version_string) { "1.0.0-Alpha" }
+
+      it { is_expected.to be(true) }
+    end
+
+    context "with a dev token" do
+      let(:version_string) { "1.2.1-dev-65" }
+
+      it { is_expected.to be(true) }
+    end
+
+    context "with a 'pre' pre-release separated with a -" do
+      let(:version_string) { "2.10.0-pre0" }
+
+      it { is_expected.to be(true) }
+    end
+
+    context "with a release" do
+      let(:version_string) { "1.0.0" }
+
+      it { is_expected.to be(false) }
+    end
+
+    context "with a + separated alphanumeric build identifier" do
+      let(:version_string) { "1.0.0+build1" }
+
+      it { is_expected.to be(false) }
+    end
+
+    context "with an 'alpha' separated by a -" do
+      let(:version_string) { "1.0.0-alpha+001" }
+
+      it { is_expected.to be(true) }
+    end
+  end
+
+  describe "#<=>" do
+    it "sorts version strings semantically" do
+      versions = []
+
+      versions << described_class.new("1.0.0-alpha")
+      versions << described_class.new("1.0.0-alpha.1")
+      versions << described_class.new("1.0.0-alpha.1.beta.gamma")
+      versions << described_class.new("1.0.0-alpha.beta")
+      versions << described_class.new("1.0.0-alpha.beta.1")
+      versions << described_class.new("1.0.0-beta")
+      versions << described_class.new("1.0.0-beta.2")
+      versions << described_class.new("1.0.0-beta.11")
+      versions << described_class.new("1.0.0-rc.1")
+      versions << described_class.new("1.0.0")
+      expect(versions.shuffle.sort).to eq(versions)
```

Comment:
Can we have more tests to cover the other methods in the base class please? Also, what's the current test coverage?

Reply:
96.55% It's got the highest coverage of all the files in dependabot common.